addon/PreviewProvider.js

```diff
@@ -355,6 +356,11 @@ PreviewProvider.prototype = {
     this._metadataEndpoints.set("Embedly", simplePrefs.prefs["embedly.endpoint"]);
     this.enabled = simplePrefs.prefs[ENABLED_PREF];
     simplePrefs.on("", this._onPrefChange);
+
+    // if we are in the experiment change the metadata source
+    if (this._experimentProvider.data.metadataService) {
+      simplePrefs.prefs[METADATA_SOURCE_PREF] = "MetadataService";
```

Comment:
Let's move all the hard coded strings of metadata source names to constants and refer to them that way.